common/management/commands/init_db.py

```diff
@@ -0,0 +1,239 @@
+from dataclasses import dataclass
+from django.core.management.base import BaseCommand
+from django.contrib.auth.models import Group
+from common.models import Class, Subject, User, Semester, Task, AssignedTask
+from django.utils import timezone
+import datetime
+import os
+import random
+import string
+import logging
+
+
+def create_user(username: str, is_teacher: bool) -> User:
+    user, created = User.objects.get_or_create(username=username, email="")
+    if created:
+        user.set_password("admin")
+    if is_teacher:
+        teachers_group = Group.objects.get(name="teachers")
+        teachers_group.user_set.add(user)
+    user.save()
+    return user
+
+
+def create_semesters(date: datetime, num_semesters: int) -> list[Semester]:
+    semesters = []
+    for i in range(num_semesters):
+        summer = 1 < date.month < 7
+        begin = datetime.datetime.strptime(
+            f"{date.year}-02-01" if summer else f"{date.year}-10-01", "%Y-%m-%d"
+        ).date()
+        end = datetime.datetime.strptime(
+            f"{date.year}-07-31" if summer else f"{date.year + 1}-01-31", "%Y-%m-%d"
+        ).date()
+        semester, _ = Semester.objects.get_or_create(
+            begin=begin,
+            end=end,
+            year=f"{date.year}",
+            winter=not summer,
+            active=i == 0,
+            inbus_semester_id=124 + i,
+        )
+        semesters.append(semester)
+        if summer:
+            date = date.replace(month=8)
+            date = date.replace(year=date.year - 1)
+        else:
+            date = date.replace(month=3)
+    return semesters
+
+
+def create_subject(name: str, abbr: str) -> Subject:
+    subj, _ = Subject.objects.get_or_create(name=name, abbr=abbr)
+    return subj
+
+
+def create_class(
+    code: str, teacher: User, semester: Semester, subject: Subject, day: Class.Day, time: datetime
+) -> Class:
+    created_class, _ = Class.objects.get_or_create(
+        code=code, teacher=teacher, semester=semester, subject=subject, day=day, time=time
+    )
+    return created_class
+
+
+def create_random_class(teacher: User, semesters: list[Semester], subjects: list[Subject]) -> Class:
+    code = f"{random.choice(string.ascii_uppercase)}/{''.join(random.choices(string.digits, k=2))}"
+    chosen_semester = random.choice(semesters)
```

Comment:
If the RNG is unlucky, the person could end up with no classes in the currently active semester :) FWIW, I don't think we even need to do multiple semesters. Bootstrapping a single semester is enough, and it should simplify the code considerably. We could make the semester year/summer be a parameter, if someone needed to overwrite it.